code/js/rcalc-localization.js

```diff
@@ -0,0 +1,25 @@
+var set_locale_to = function(locale) {
+  if (locale) {
+    $.i18n().locale = locale;
+  }
+  $('title').i18n();
+  $('body').i18n();
+};
+
+jQuery(function() {
+  $.i18n().load({
+    'en': '../js/i18n/en.json',
+    'cs': '../js/i18n/cs.json'
+  }).done(function() {
+    set_locale_to(url('?locale'));
+
+    History.Adapter.bind(window, 'statechange', function() {
+      set_locale_to(url('?locale'));
+    });
+
+    $('.rcalc-dropdown-content-language').on('click', 'a', function(e) {
+      e.preventDefault();
+      History.pushState(null, null, "?locale=" + $(this).data('locale'));
```

Comment:
String ?locale is used two times in this file. What should be done?

Reply:
string ?locale has been replaced by constant QUESTION_MARK_LOCALE

Reply:
And why you don't use constant in second method: History.pushState(...
Btw. Sometimes is enough create constants just in file where is constant used. On the start project we created way create constant especially in separate files but don't worry create also in file where it is used. In separate file it especially used when constant is used in more than one file. There it make sense ... but doesn't mind. We will see what happens in future.

Reply:
created const LOCALE = "locale", const QUESTION_MARK = "?", const EQUALS_SIGN = "=" and now we have constans everywhere where locale is present